modules/collect/src/main/java/com/opengamma/collect/Messages.java

```diff
@@ -18,6 +18,44 @@ private Messages() {
 
   //-------------------------------------------------------------------------
   /**
+   * Formats a templated message inserting a single argument.
+   * <p>
+   * This method combines a template message with a single argument.
+   * It can be useful to delay string concatenation, which is sometimes a performance issue.
+   * The approach is similar to SLF4J MessageFormat, Guava Preconditions and String format().
+   * <p>
+   * The message template contains zero to many "{}" placeholders.
+   * The first placeholder is replaced by the string form of the argument.
+   * Subsequent placeholders are not replaced.
+   * If there is no placeholder, then the argument is appended to the end of the message.
+   * No attempt is made to format the argument.
+   * <p>
+   * This method is null tolerant to ensure that use in exception construction will
+   * not throw another exception, which might hide the intended exception.
+   * 
+   * @param messageTemplate  the message template with "{}" placeholders, null returns empty string
+   * @param arg  the message argument, null treated as string "null"
+   * @return the formatted message
+   */
+  public static String format(String messageTemplate, Object arg) {
+    if (messageTemplate == null) {
+      return format("", arg);
+    }
+    int placeholderPos = messageTemplate.indexOf("{}", 0);
+    String argStr = String.valueOf(arg);
+    StringBuilder builder = new StringBuilder(messageTemplate.length() + argStr.length());
```

Comment:
This builder is the right size if the template includes ``{}`` but too small if the code goes into the ``else`` branch. Do we care?